### PR TITLE
Toggle folding on same buffer

### LINKED ISSUE #169: Enhancements to SignifyFold
Using fold on changes is great, feels like you are in front of a unified diff output, but editable. just a few tweaks:

- Using fold seems to use `:only` hidding all other buffers opened
- Activating SignifyToggleHighlight does not refresh colours until you pass cursor from changed lines
- What about `:SignifyToggleFold`. You can use `zR`, `zM` to open close but `zE` doesn't work.

### PR DESCRIPTION
**Pull request by albfan**
resolves #169 

Sorry, I can resist! Close it inmediately if you want.

### CODE REVIEW

**Comment by mhinz** [comment]
Well, the patch assumed that the user was using manual folding previously. ;]

But that's easy to fix. You can get the value of options like this:
```viml
let old_fdm = &foldmethod
setlocal foldmethod=expr
" do stuff
let &foldmethod = old_fdm
```
The real problem is, assuming the user used manual folding, all folds get lost. To work around this your best bet is to use `:mkview`.

You can change your PR if you want, otherwise I'll whip up a patch later. This seems useful enough. :-)

**Comment by albfan** [comment]
I checked using

    set fdm=indent

and works pretty well. should be some way to recover previous created folds, anyway we can live without that as you say. I promise to do some research on that.

**Comment by jamessan** [comment]
I find the pattern of `open a new window/tab page to display a different view of the buffer` very useful, so I'd dislike to lose that.  Maybe have `SignifyToggleFold` open in a new tab page and `SignifyToggleFold!` modify the view in the current window.

**Comment by mhinz** [comment]
@jamessan :+1:

@albfan If you don't find a good solution other than `:mkview` for restoring folds, don't worry. I'll use your two commits anyway, add an additional commit for the `:SignifyToggleFold`/`:SignifyToggleFold!` distinction, and merge it into master.

I'll check back on this PR in some hours; Christmas market in the meanwhile!

**Comment by albfan** [comment]
Great! I suppose others would prefer open a tab, so that is a good solution. I'm not totally satisfied with detection of folding style on disable, see code review.

File: plugin/signify.vim [hunk 1]
```diff
@@ -38,13 +38,14 @@ augroup END
...
-command! -nargs=0 -bar SignifyToggle          call sy#toggle()
-command! -nargs=0 -bar SignifyToggleHighlight call sy#highlight#line_toggle()
+command! -nargs=0 -bar       SignifyDebug           call sy#debug#list_active_buffers()
+command! -nargs=0 -bar       SignifyDebugDiff       call sy#debug#verbose_diff_cmd()
+command! -nargs=0 -bar       SignifyDebugUnknown    call sy#repo#debug_detection()
+command! -nargs=0 -bar       SignifyFold            call sy#fold#do()
+command! -nargs=0 -bar       SignifyRefresh         call sy#util#refresh_windows()
+command! -nargs=0 -bar       SignifyToggle          call sy#toggle()
+command! -nargs=0 -bar       SignifyToggleHighlight call sy#highlight#line_toggle()
+command! -nargs=0 -bar -bang SignifyToggleFold      call sy#fold#toggle(<q-bang>)
```
Comment (jamessan) [inline plugin/signify.vim]: A more typical way of doing this is `call sy#fold#toggle(<bang>0)`.  Then the argument to `sy#fold#toggle` can be named `newtab`, which will have a value of 0 when `SignifyToggleFold` is called and 1 when `SignifyToggleFold!` is called.

File: plugin/signify.vim [hunk 2]
```diff
@@ -38,13 +38,14 @@ augroup END
...
-command! -nargs=0 -bar SignifyDebugDiff       call sy#debug#verbose_diff_cmd()
-command! -nargs=0 -bar SignifyDebugUnknown    call sy#repo#debug_detection()
-command! -nargs=0 -bar SignifyFold            call sy#fold#do()
-command! -nargs=0 -bar SignifyRefresh         call sy#util#refresh_windows()
-command! -nargs=0 -bar SignifyToggle          call sy#toggle()
-command! -nargs=0 -bar SignifyToggleHighlight call sy#highlight#line_toggle()
+command! -nargs=0 -bar       SignifyDebug           call sy#debug#list_active_buffers()
+command! -nargs=0 -bar       SignifyDebugDiff       call sy#debug#verbose_diff_cmd()
+command! -nargs=0 -bar       SignifyDebugUnknown    call sy#repo#debug_detection()
+command! -nargs=0 -bar       SignifyFold            call sy#fold#do()
```
Comment (jamessan) [inline plugin/signify.vim]: This should also have `-bang` now.

**Comment by mhinz** [comment]
@albfan I merged your commits and added some additional commits. *cough*

`:SignifyFold` is untouched and `:SignifyFold!` should toggle in the current window for all possible settings of `'foldexpr'`.

**Pull request merged**